server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceShardsAllocator.java

```diff
@@ -281,8 +286,6 @@ public void onFailure(Exception e) {
             if (e.getCause() != null && e.getCause() instanceof EsRejectedExecutionException esRejectedExecutionException) {
                 assert esRejectedExecutionException.isExecutorShutdown();
                 // TODO now what? onNoLongerMaster() asserts it's on the master thread but we could be anywhere here
```

Comment:
I reckon we can drop all of this now. Certainly the comment is unnecessary, but nor do we really need to branch here.

Reply:
Sorry, not sure I follow. Do you want to drop `assert esRejectedExecutionException.isExecutorShutdown();` assertion?

Reply:
Yes, I don't think that assertion is particularly important.